examples/dlint/main.rs

```diff
@@ -124,23 +124,33 @@ fn display_diagnostics(
   }
 }
 
-fn run_linter<F>(paths: Vec<String>, get_rules: F)
-where
-  F: Fn() -> Vec<Box<dyn LintRule>> + Sync,
-{
+fn run_linter(paths: Vec<String>, filter_rule_name: Option<&str>) {
   let error_counts = Arc::new(AtomicUsize::new(0));
   let output_lock = Arc::new(Mutex::new(())); // prevent threads outputting at the same time
 
   paths.par_iter().for_each(|file_path| {
     let source_code =
       std::fs::read_to_string(&file_path).expect("Failed to read file");
 
-    let mut linter = LinterBuilder::default().rules(get_rules()).build();
+    let mut linter = LinterBuilder::default()
+      .rules(get_recommended_rules())
```

Comment:
Why not filter rule list before running linter?

Reply:
I did that way first, but I realized that doing so led to unexpected result. That said, consider the case where we use the code of 836c2ff239464fb3c14f592dff6cb7894e65d0fd and the target script is:

```ts
// deno-lint-ignore no-explicit-any
let foo: any = 1;
console.log(foo);
```

Running `cargo run --example dlint -- run --rule prefer-const hoge.ts`, then we get:

```
error[ban-unknown-rule-code]: Unknown rule for code "no-explicit-any"
 --> hoge.ts:1:1
  |
1 | // deno-lint-ignore no-explicit-any
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
error[prefer-const]: 'foo' is never reassigned. Use 'const' instead
 --> hoge.ts:2:5
  |
2 | let foo: any = 1;
  |     ^^^
  |
Found 2 problems
```

This shows that the linter treats rules other than the specified one in `--rule` as "unknown" rule. To address this problem, I moved filtering a rule down.

Also I think we could address it by fixing `filter_diagnostics` method in `linter.rs`, though I'm not sure which way is better.

Reply:
Ah I see, that makes sense